src/dispatch/__tests__/Authentication.test.ts

```diff
@@ -336,4 +336,69 @@ describe('Authentication tests', () => {
             })
         );
     });
+
+    test('when credentials are read from storage then a new date object is created', async () => {
+        // Init
+        const storageExpiration = new Date(Date.now() + 3600 * 1000);
+
+        localStorage.setItem(
+            CRED_KEY,
+            JSON.stringify({
+                accessKeyId: 'a',
+                secretAccessKey: 'b',
+                sessionToken: 'c',
+                expiration: storageExpiration
+            })
+        );
+
+        const config = {
+            ...DEFAULT_CONFIG,
+            ...{
+                identityPoolId: IDENTITY_POOL_ID,
+                guestRoleArn: GUEST_ROLE_ARN
+            }
+        };
+        const auth = new Authentication(config);
+
+        // Run
+        const credentials = await auth.ChainAnonymousCredentialsProvider();
+
+        // Assert
+        expect(credentials.expiration!.getTime()).toEqual(
+            storageExpiration.getTime()
+        );
+    });
+
+    test('when credentials are read from storage then the member variable stores the expiration as a date object', async () => {
```

Comment:
nit: This test will pass under certain conditions when it should fail (i.e., false negative).

Specifically, when `this.credentials.expiration` is `undefined`, and the `expiration` property of the value returned by `AnonymousStorageCredentialsProvider` is a `Date` object. When this happens, control flow will reach `AnonymousCognitoCredentialsProvider`, and the test needs a way to distinguish the value returned here from the expected value.

You can fix this by mocking the response of getCredentials, like was done in [EnhancedAuthentication.test.ts here](https://github.com/aws-observability/aws-rum-web/pull/436/files#diff-94a33512cc69b461d93e786b0cfc32e12f8767a11831604e38300860eb1ffba7R348-R355).